staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -214,17 +214,38 @@ func newETCD3Check(c storagebackend.Config, timeout time.Duration, stopCh <-chan
 		}
 		ctx, cancel := context.WithTimeout(context.Background(), timeout)
 		defer cancel()
-		// See https://github.com/etcd-io/etcd/blob/c57f8b3af865d1b531b979889c602ba14377420e/etcdctl/ctlv3/command/ep_command.go#L118
 		now := time.Now()
-		_, err := client.Get(ctx, path.Join("/", c.Prefix, "health"))
-		if err != nil {
-			err = fmt.Errorf("error getting data from etcd: %w", err)
-		}
+		err := prober.Probe(ctx)
 		lastError.Store(err, now)
 		return err
 	}, nil
 }
 
+func newETCD3Prober(c storagebackend.Config) (*etcd3Prober, error) {
+	client, err := newETCD3Client(c.Transport)
+	if err != nil {
+		return nil, err
+	}
+	return &etcd3Prober{
+		Client: client,
+		prefix: c.Prefix,
+	}, nil
+}
+
+type etcd3Prober struct {
+	*clientv3.Client
+	prefix string
+}
+
+func (p *etcd3Prober) Probe(ctx context.Context) error {
+	// See https://github.com/etcd-io/etcd/blob/c57f8b3af865d1b531b979889c602ba14377420e/etcdctl/ctlv3/command/ep_command.go#L118
+	_, err := p.Client.Get(ctx, path.Join("/", p.prefix, "health"))
```

Comment:
Close should close the underlying grpc connection. I expect the Probe to cause panic.

Reply:
we can not panic inside the apiserver, we may have a boolean to indicate we are stopped or nil the client on close ... so we can return an error on a Probe instead of panicing

Reply:
@wojtek-t do we want to have this risk of panicing?

Reply:
We don't - I somehow missed that comment - thanks for catching.

That said, I'm not sure I fully follow your concern.
If a client will get closed, there are multiple other places in the code that would panic. 

`CreateProber` is called only from a single place and in that place we know that it's not the case.
So we're safe now (unless I'm missing something - please let me know).

So you're rather concerned about future use of that, right?

Reply:
right, since we are now exposing it I can see how someone in the future may try to. Probe() in a closed state

Reply:
OK - so at least I didn't miss something for now.

But I agree - the changes you proposed (together with making etcd3.Client a private field instead) are simple enough and would already protect us - so I'm definitely in favor.

@serathius - can you please apply?

Reply:
Done, even made it thread safe.

Reply:
Not really - see my comment above.